io-engine/Cargo.toml

```diff
@@ -8,6 +8,7 @@ default-run = "io-engine-client"
 build = "build.rs"
 
 [features]
+default = ["fault_injection"]
```

Comment:
Do we want this built by default?

Reply:
Not exactly, but we don't have an option to disable it for now: too much work required to modify testing pipeline to enable running this particular test with a special build.

I added a global boolean variable to check if any injections present. So when no injections, it is just an access to atomic bool per I/O operation. 

Once we have a support for a special builds for specific tests, we'll disable this by default.

Reply:
Should we change the nix build to disable fault_injection when building release?